### ISSUE TITLE
takeRight can result in OutOfMemoryExceptions being thrown

### ISSUE THREAD

**Issue by johnmcclean** [aol/cyclops#1013]
e.g. 
```java
assertThat(empty.takeRight(Integer.MAX_VALUE),equalTo(empty));
```

The underlying cause is the use of takeRight on ReactiveSeq which allocates an ArrayDeque of the specified size to capture the last X values. Sizing the ArrayDeque to a sensible max and switching to a LinkedList above that max would solve this issue.

**Comment by johnmcclean** [comment]
Related to #1012

**Comment by johnmcclean** [comment]
Merged in #1016
